File: 73. Set Matrix Zeroes/setZeroes.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        int row = matrix.size();
        int col = matrix[0].size();

        vector<vector<int>> zeros;

        for(int r=0 ; r<row; r++){

            for(int c = 0 ; c<col ; c++){

                if(matrix[r][c] == 0)zeros.push_back({r,c});
            }
        }

        for(auto position : zeros){
            int r = position[0];
            int c = position[1];

            for(int i=0 ; i<row; i++){
                matrix[i][c] = 0;
            }

            for(int j=0 ; j<col; j++){
                matrix[r][j] = 0;
            }
        }
    }
};

File: 73. Set Matrix Zeroes/setZeroes.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var setZeroes = function (matrix) {
    let row = matrix.length
    let col = matrix[0].length

    let zeros = []

    for (let i = 0; i < row; i++) {
        for (let j = 0; j < col; j++) {

            if (matrix[i][j] === 0) zeros.push([i, j])
        }
    }

    for (let pos of zeros) {
        let [r, c] = pos
        for (let i = 0; i < row; i++) {
            matrix[i][c] = 0
        }
        for (let j = 0; j < col; j++) {
            matrix[r][j] = 0
        }
    }
};
